src/deque.jl

```diff
@@ -196,3 +196,4 @@ end
 
 # TODO proper multidimension boundscheck
 @propagate_inbounds setindex(a::StaticArray, x, inds::Int...) = setindex(a, x, LinearIndices(a)[inds...])
+@propagate_inbounds setindex(a::StaticArray, x, I::CartesianIndex) = setindex(a, x, I.I...)
```

Comment:
Not a big deal, but I guess `I.I` is an implementation detail?
```suggestion
@propagate_inbounds setindex(a::StaticArray, x, I::CartesianIndex) = setindex(a, x, Tuple(I)...)
```

Reply:
This is now irrelevant with the change below.